momepy/functional/_intensity.py

```diff
@@ -111,3 +111,56 @@ def _calc_nodedensity(group, edges):
         summation_values = pd.Series(np.ones(nodes.shape[0]), index=nodes.index)
 
     return graph.apply(summation_values, _calc_nodedensity, edges=edges)
+
+
+def density(
+    values: Series | np.ndarray, areas: Series | np.ndarray, graph: Graph
+) -> Series:
+    """Calculate the gross density.
+
+    .. math::
+        \\frac{\\sum \\text {values}}{\\sum \\text {areas}}
+
+    Adapted from :cite:`dibble2017`.
+
+    Parameters
+    ----------
+    values : pd.Series | np.ndarray
+        The character values for density calculations.
+        The index is used to arrange the final results.
+    areas : np.array | pd.Series
+        The area values for the density calculations,
+        an ``np.ndarray``, or ``pd.Series``.
+    graph : libpysal.graph.Graph
+        A spatial weights matrix for the geodataframe,
+        it is used to denote adjacent elements.
+
+    Returns
+    -------
+    DataFrame
+
+
+    Examples
+    --------
+    >>> tessellation_df['floor_area_dens'] = mm.density(tessellation_df['floor_area'],
+    ...                                                 tessellation_df['area'],
+    ...                                                 graph)
+    """
+
+    if isinstance(values, np.ndarray):
+        values = pd.DataFrame(values)
+    elif isinstance(values, pd.Series):
+        values = values.to_frame()
+
+    if isinstance(areas, np.ndarray):
+        areas = pd.Series(values)
+
+    stats = graph.apply(
+        pd.concat((values, areas.rename("area")), axis=1),
+        lambda x: (x.loc[:, x.columns != "area"].sum() / x["area"].sum()),
```

Comment:
are you assuming here that `values` can be 2d and calculate density for multiple vars at one go? That is undocumented and goes against typing.